device/common/include/traccc/device/fill_prefix_sum.hpp

```diff
@@ -0,0 +1,39 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021-2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/definitions/qualifiers.hpp"
+#include "traccc/device/get_prefix_sum.hpp"
+
+// VecMem include(s).
+#include <vecmem/memory/memory_resource.hpp>
+#include <vecmem/utils/copy.hpp>
+
+namespace traccc::device {
+
+/// Function filling the prefix sum for a "size vector"
+///
+/// This simple function is just meant to translate the received "size vector"
+/// into a "prefix sum". I.e. into a list of index pairs that would allow
+/// visiting all elements of the jagged vector described by this "size vector".
+///
+/// @param[in] globalIndex The index of the current thread
+/// @param[in] sizes_view View on the sizes of the "inner vectors" of a jagged
+/// vector
+/// @param[out] ps_view   View on the result vector of index pairs
+///
+TRACCC_HOST_DEVICE
+void fill_prefix_sum(unsigned int globalIndex,
+                     const vecmem::data::vector_view<unsigned int>& sizes_view,
```

Comment:
This will be fairly anal, but... Let's not hardcode `unsigned int` here. Let's rather write:

```c++
const vecmem::data::vector_view<vecmem::device_vector<int>::size_type>& sizes_view
```

instead. So that if at any point we may change how `vecmem::device_vector` represents its size, we wouldn't need to change the code here.

To help client code a bit, you could introduce a shorter typedef here. Like:

```c++
namespace traccc::device {
    using prefix_sum_size_t = vecmem::device_vector<int>::size_type;
}
```

:thinking:

Also note that since this is an input variable, the type should actually be `const`. Like:

```c++
const vecmem::data::vector_view<const prefix_sum_size_t>& sizes_view
```